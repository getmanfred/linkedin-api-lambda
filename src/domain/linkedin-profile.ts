export class LinkedinProfileProfile {
  public 'First Name'?: string;
  public 'Last Name'?: string;
  public 'Headline'?: string;
  public 'Summary'?: string;
}

export class LinkedinProfileSkill {
  public 'Name'?: string;
}

export class LinkedinProfilePosition {
  public 'Title'?: string;
  public 'Description'?: string;
  public 'Company Name'?: string;
  public 'Started On'?: string;
  public 'Finished On'?: string;
}

export class LinkedinProfileEducation {
  public 'School Name'?: string;
  public 'Degree Name'?: string;
  public 'Start Date'?: string;
  public 'End Date'?: string;
}

// -- ðŸ‘‰ main class
export class LinkedinProfile {
  public profile: LinkedinProfileProfile;
  public skills: LinkedinProfileSkill[];
  public positions: LinkedinProfilePosition[];
  public education: LinkedinProfileEducation[];

  public constructor() {
    this.profile = new LinkedinProfileProfile();
    this.skills = [];
    this.positions = [];
    this.education = [];
  }
}
