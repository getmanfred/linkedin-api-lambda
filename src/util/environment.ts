/* eslint-disable no-process-env */
import { bool, cleanEnv, str } from 'envalid';
import { LinkedinProfileRequest } from '../contracts/linkedin-profile.request';
import { EnvironmentType } from '../domain/logger-context';
import { setContextLogger } from './logger';

export class Environment {
  public readonly LOGGER_CONSOLE: boolean;
  public readonly LOCAL_LINKEDIN_API_TOKEN?: string;

  private constructor(_envName: EnvironmentType) {
    const env = cleanEnv(process.env, {
      LOGGER_CONSOLE: bool({ default: false }),
      LOCAL_LINKEDIN_API_TOKEN: str({ default: undefined })
    });

    this.LOGGER_CONSOLE = env.LOGGER_CONSOLE;
    this.LOCAL_LINKEDIN_API_TOKEN = env.LOCAL_LINKEDIN_API_TOKEN ?? undefined;
  }

  public static setupEnvironment(request: LinkedinProfileRequest): Environment {
    const environment = new Environment(request.env);
    setContextLogger({
      messageId: request.messageId,
      contextId: request.contextId,
      profileId: request.profileId,
      env: request.env
    });
    return environment;
  }
}
