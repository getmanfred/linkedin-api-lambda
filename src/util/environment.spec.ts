/* eslint-disable no-process-env */
import { createMockedLinkedinProfileRequest } from '../test/mocks/linkedin-profile.mocks';
import { Environment } from './environment';

describe('Environment config', () => {
  const mockEnv = {
    MAX_RETRIES: '5',
    LOGGER_CONSOLE: 'true',
    LOCAL_LINKEDIN_API_TOKEN: 'mock-api-token',
    AWS_REGION: 'us-east-1',
    AWS_SQS_ENDPOINT: 'http://localhost:4566',
    AWS_QUEUE_URL_STAGE: 'https://sqs-stage.amazonaws.com/1234567890/stage-queue',
    AWS_RESULT_QUEUE_URL_STAGE: 'https://sqs-stage.amazonaws.com/1234567890/stage-result-queue',
    AWS_QUEUE_URL_PRO: 'https://sqs-pro.amazonaws.com/1234567890/pro-queue',
    AWS_RESULT_QUEUE_URL_PRO: 'https://sqs-pro.amazonaws.com/1234567890/pro-result-queue'
  };

  beforeEach(() => {
    process.env = { ...mockEnv };
  });

  afterEach(() => {
    jest.resetModules();
  });

  it('should correctly load environment variables for "stage" environment and match snapshot', () => {
    const request = createMockedLinkedinProfileRequest({ env: 'stage' });

    const environment = Environment.setupEnvironment(request);

    expect(environment).toMatchSnapshot();
  });

  it('should correctly load environment variables for "pro" environment and match snapshot', () => {
    const request = createMockedLinkedinProfileRequest({ env: 'pro' });

    const environment = Environment.setupEnvironment(request);

    expect(environment).toMatchSnapshot();
  });

  it('should throw an error when environment variables are invalid', () => {
    const mockExit = jest.spyOn(process, 'exit').mockImplementation(() => {
      throw new Error('process.exit called');
    });

    process.env.MAX_RETRIES = 'invalid';

    const request = createMockedLinkedinProfileRequest({ env: 'pro' });

    expect(() => Environment.setupEnvironment(request)).toThrowErrorMatchingSnapshot();

    mockExit.mockRestore();
  });
});
