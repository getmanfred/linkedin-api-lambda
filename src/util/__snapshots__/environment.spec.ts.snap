// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment config should correctly load environment variables for "pro" environment and match snapshot 1`] = `
Environment {
  "AWS_QUEUE_URL": "https://sqs-pro.amazonaws.com/1234567890/pro-queue",
  "AWS_REGION": "us-east-1",
  "AWS_RESULT_QUEUE_URL": "https://sqs-pro.amazonaws.com/1234567890/pro-result-queue",
  "AWS_SQS_ENDPOINT": "http://localhost:4566",
  "LOCAL_LINKEDIN_API_TOKEN": "mock-api-token",
  "LOGGER_CONSOLE": true,
  "MAX_RETRIES": 5,
}
`;

exports[`Environment config should correctly load environment variables for "stage" environment and match snapshot 1`] = `
Environment {
  "AWS_QUEUE_URL": "https://sqs-stage.amazonaws.com/1234567890/stage-queue",
  "AWS_REGION": "us-east-1",
  "AWS_RESULT_QUEUE_URL": "https://sqs-stage.amazonaws.com/1234567890/stage-result-queue",
  "AWS_SQS_ENDPOINT": "http://localhost:4566",
  "LOCAL_LINKEDIN_API_TOKEN": "mock-api-token",
  "LOGGER_CONSOLE": true,
  "MAX_RETRIES": 5,
}
`;

exports[`Environment config should throw an error when environment variables are invalid 1`] = `"ðŸ”§ [Environment] Invalid environment variables: process.exit called"`;
